cmd/cloud/installation.go

```diff
@@ -1,15 +1,178 @@
+	installationCreateCmd.Flags().String("version", "stable", "The Mattermost version to install.")
+	installationCreateCmd.Flags().String("dns", "", "The URL at which the Mattermost server will be available.")
+	installationCreateCmd.Flags().String("affinity", "isolated", "How other installations may be co-located in the same cluster.")
+	installationCreateCmd.MarkFlagRequired("owner")
+	installationCreateCmd.MarkFlagRequired("dns")
+
+	installationUpgradeCmd.Flags().String("installation", "", "The id of the installation to be upgraded.")
+	installationUpgradeCmd.Flags().String("version", "stable", "The Mattermost version to target.")
+	installationUpgradeCmd.MarkFlagRequired("installation")
+	installationUpgradeCmd.MarkFlagRequired("version")
+
+	installationDeleteCmd.Flags().String("installation", "", "The id of the installation to be deleted.")
+	installationDeleteCmd.MarkFlagRequired("installation")
+
+	installationGetCmd.Flags().String("installation", "", "The id of the installation to be fetched.")
+	installationGetCmd.MarkFlagRequired("installation")
+
+	installationListCmd.Flags().String("owner", "", "The owner by which to filter installations.")
+	installationListCmd.Flags().Int("page", 0, "The page of installations to fetch, starting at 0.")
+	installationListCmd.Flags().Int("per-page", 100, "The number of installations to fetch per page.")
+	installationListCmd.Flags().Bool("include-deleted", false, "Whether to include deleted installations.")
+
+	installationCmd.AddCommand(installationCreateCmd)
+	installationCmd.AddCommand(installationUpgradeCmd)
+	installationCmd.AddCommand(installationDeleteCmd)
+	installationCmd.AddCommand(installationGetCmd)
+	installationCmd.AddCommand(installationListCmd)
+}
+
 var installationCmd = &cobra.Command{
 	Use:   "installation",
 	Short: "Manipulate installations managed by the provisioning server.",
-	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("TODO: implement installation command")
+}
+
+var installationCreateCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create an installation.",
+	RunE: func(command *cobra.Command, args []string) error {
+		command.SilenceUsage = true
+
+		serverAddress, _ := command.Flags().GetString("server")
+		client := api.NewClient(serverAddress)
+
+		ownerID, _ := command.Flags().GetString("owner")
+		version, _ := command.Flags().GetString("version")
+		dns, _ := command.Flags().GetString("dns")
+		affinity, _ := command.Flags().GetString("affinity")
+
+		installation, err := client.CreateInstallation(&api.CreateInstallationRequest{
+			OwnerID:  ownerID,
+			Version:  version,
+			DNS:      dns,
+			Affinity: affinity,
+		})
+		if err != nil {
+			return errors.Wrap(err, "failed to create installation")
+		}
+
+		err = printJSON(installation)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+var installationUpgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrade (or downgrade) the version of Mattermost.",
```

Comment:
5/5 that we should support downgrades: otherwise there is no way to rollback an installation gone awry. The current plan is to let the operator decide how far back to allow such a downgrade -- e.g. we may not support downgrading across major versions.